### PR TITLE
Add documentation for BigQuery transfer operators

### LINKED ISSUE #27093: Missing documentation for GCP BigQuery transfer operators
### What do you see as an issue?

The transfer operators from and to Google Cloud BigQuery seem to be currently missing in the documentation.

Current documentation on Google Cloud transfer operators: https://airflow.apache.org/docs/apache-airflow-providers-google/stable/operators/transfer/index.html

List of the possible missing operators in the documentation: BigQueryToBigQueryOperator, BigQueryToGCSOperator, BigQueryToMsSqlOperator, BigQueryToMySqlOperator, GCSToBigQueryOperator

List of available GCP transfer operators: https://github.com/apache/airflow/tree/main/airflow/providers/google/cloud/transfers

Could someone please confirm if the operators are indeed missing in the documentation? I wasn't able to find them elsewhere, and it seems like they should at least be referenced there. 

### Solving the problem

I would gladly submit a PR to help document the five missing operators. If the documentation is indeed missing, is it okay to use a single PR for all five operators, or should we separate them?

### Anything else

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by nivaldoh**
closes: #27093

Adds documentation for the following Google Cloud transfer operators: BigQueryToBigQueryOperator, BigQueryToGCSOperator, BigQueryToMsSqlOperator, BigQueryToMySqlOperator, GCSToBigQueryOperator

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: docs/apache-airflow-providers-google/operators/cloud/gcs.rst [hunk 1]
```diff
@@ -32,22 +32,6 @@ Prerequisite Tasks
...
-:class:`~airflow.providers.google.cloud.transfers.gcs_to_bigquery.GCSToBigQueryOperator`
-to execute a BigQuery load job.
-
-.. exampleinclude:: /../../tests/system/providers/google/cloud/gcs/example_gcs_to_bigquery.py
-    :language: python
-    :dedent: 4
-    :start-after: [START howto_operator_gcs_to_bigquery]
-    :end-before: [END howto_operator_gcs_to_bigquery]
-
-
```
Comment (nivaldoh) [inline docs/apache-airflow-providers-google/operators/cloud/gcs.rst]: Migrate the GCSToBigQueryOperator to its own page to maintain the established pattern for transfer operators.

File: airflow/providers/google/cloud/transfers/bigquery_to_mysql.py [hunk 2]
```diff
@@ -45,13 +48,15 @@ class BigQueryToMySqlOperator(BaseOperator):
...
-
+    
+       # [START howto_operator_bigquery_to_mysql]
        transfer_data = BigQueryToMySqlOperator(
             task_id='task_id',
             dataset_table='origin_bq_table',
             mysql_table='dest_table_name',
             replace=True,
         )
+        # [END howto_operator_bigquery_to_mysql]
```
Comment (nivaldoh) [inline airflow/providers/google/cloud/transfers/bigquery_to_mysql.py]: Use example in the docstring as BigQueryToMySqlOperator doesn't have its own system test, and creating one may not be necessary.

**Comment by nivaldoh** [comment]
@turbaszek @mik-laj Ready for review at your earliest convenience.

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
Let's see if CI passes.

**Comment by potiuk** [comment]
Some static checks are failing - you need to solve and rebase.

**Comment by nivaldoh** [comment]
Thanks for the review. All points have been addressed.

**Comment by potiuk** [comment]
docs are failing.

**Comment by potiuk** [comment]
?

**Comment by eladkal** [comment]
@nivaldoh can you please address the doc test failures?

**Comment by shahar1** [comment]
@eladkal Please assign me

**Comment by potiuk** [comment]
> @eladkal Please assign me

You can just take the PR and start working on yours (and make reference to this PR). There is no need to assign it.

**Comment by eladkal** [comment]
resolved in https://github.com/apache/airflow/pull/29466

**Pull request closed without merging**